### PR TITLE
[Docs] Updated docs for tf.custom_gradient with examples

### PR DESCRIPTION
**Pull request by Ghost---Shadow**
There is no proper explanation of `dy` in the docs. So, I added [that](https://stackoverflow.com/questions/63146831/what-is-the-analytic-interpretation-for-tensorflow-custom-gradient/63187019#63187019).

Also added an example for [multiple inputs](https://stackoverflow.com/a/57379955/1217998)

### CODE REVIEW

File: tensorflow/python/ops/custom_gradient.py [hunk 1]
```diff
@@ -125,6 +125,45 @@ def grad(dy):
...
+  with tf.GradientTape(persistent=True) as tape:
+      tape.watch(x)
+      tape.watch(y)
+      z = bar(x, y)
+
+  tf.print(z) # Output: 6
+  tf.print(tape.gradient(z, x)) # Output: 3
+  tf.print(tape.gradient(z, y)) # Output: 2
+  tf.print(tape.gradient(x, y)) # Output: None
+  ```
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/custom_gradient.py:165]: Let's write this as a [testable docstring](https://www.tensorflow.org/community/contribute/docs_ref#testable_docstrings)

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/custom_gradient.py (hunk 1 above)
Comment (Ghost---Shadow) [reply tensorflow/python/ops/custom_gradient.py]: Done

File: tensorflow/python/ops/custom_gradient.py [hunk 2]
```diff
@@ -125,6 +125,50 @@ def grad(dy):
...
+
+  In this case the gradient of our current function defined as `dx_i/dx_i+1 = (1 - 1 / (1 + e))`.
+  The upstream gradient `dy` would be `dx_i+1/dx_i+2 * dx_i+2/dx_i+3 * ... * dx_n/dx`.
+  The upstream gradient multiplied by the current gradient is then passed downstream.
+
+  In case the function takes multiple variables as input, the `grad` function must also return
+  the same number of variables. We take the function `z = x * y` as an example.
+
+  ```python
+  >>> import tensorflow as tf
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/custom_gradient.py:142]:
```suggestion
```

You don't need the import

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/python/ops/custom_gradient.py [hunk 3]
```diff
@@ -125,6 +125,49 @@ def grad(dy):
...
+  `dy/dx = dy/x_0 * dx_0/dx_1 * ... * dx_i/dx_i+1 * ... * dx_n/dx`
+
+  In this case the gradient of our current function defined as `dx_i/dx_i+1 = (1 - 1 / (1 + e))`.
+  The upstream gradient `dy` would be `dx_i+1/dx_i+2 * dx_i+2/dx_i+3 * ... * dx_n/dx`.
+  The upstream gradient multiplied by the current gradient is then passed downstream.
+
+  In case the function takes multiple variables as input, the `grad` function must also return
+  the same number of variables. We take the function `z = x * y` as an example.
+
+  ```python
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/custom_gradient.py:141]:
```suggestion
```

You don't need the backticks

File: tensorflow/python/ops/custom_gradient.py [hunk 4]
```diff
@@ -125,6 +125,49 @@ def grad(dy):
...
+
+  >>> z
+  6
+  >>> tape.gradient(z, x)
+  3
+  >>> tape.gradient(z, y)
+  2
+  >>> tape.gradient(x, y)
+  None
+  ```
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/custom_gradient.py:169]:
```suggestion
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by rthadur** [comment]
@Ghost---Shadow can you please check sanity build errors ?

**Comment by Ghost---Shadow** [comment]
@rthadur Please trigger the pipeline. 🤞🤞🤞

**Comment by Ghost---Shadow** [comment]
@rthadur `Linux GPU` is failing with `Not found: Container localhost does not exist. (Could not find resource: localhost/)`

https://source.cloud.google.com/results/invocations/fdcd7455-4ae4-44c7-aabd-c2c07586bdf4/targets/%2F%2Ftensorflow%2Fc%2Feager:c_api_distributed_test_gpu/log

Will re-triggering the job fix the issue?

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**